apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -44,45 +44,152 @@
        ) -> {Status :: cowboy:http_status(), Headers :: list(), Body :: map()}.
 
 handle_request('GetTopBlock', _, _Context) ->
-    {ok, TopBlock} = aehttp_logic:get_top(),
-    EncodedHash = aehttp_api_parser:encode(block_hash, TopBlock),
-    EncodedHeader = aehttp_api_parser:encode(header, TopBlock),
-    {200, [], maps:put(<<"hash">>, EncodedHash, EncodedHeader)};
+    case aec_chain:top_block() of
+        Block when Block =/= undefined ->
+            case aec_blocks:height(Block) of
+                0 ->
+                    Header = aec_blocks:to_header(Block),
+                    {200, [], aec_headers:serialize_for_client(Header, key)};
+                _ ->
+                    PrevBlockHash = aec_blocks:prev_hash(Block),
+                    case aec_chain:get_block(PrevBlockHash) of
+                        {ok, PrevBlock} ->
+                            BlockType = aec_blocks:type(Block),
+                            PrevBlockType = aec_blocks:type(PrevBlock),
+                            Header = aec_blocks:to_header(Block),
+                            Result =
+                                case BlockType of
+                                    key ->
+                                        aec_headers:serialize_for_client(Header, PrevBlockType);
+                                    micro ->
+                                        R = aec_headers:serialize_for_client(Header, PrevBlockType),
+                                        R#{<<"signature">> => aec_blocks:signature(Block)}
```

Comment:
Also it seems like microblock signature shall have some sort of encoding? (Not sure if the same of transaction signature.)

Reply:
Encoding handled in `aec_headers:serialize_for_client/2`